### DISCUSSION TITLE
How to deploy DB with FluentMigrator

### DISCUSSION THREAD

**Discussion by Alex-1357** [fluentmigrator/fluentmigrator#1497] [General] [answered]
Hello, How to create first deployment DB with FluentMigrator to my local SQL2014? I have project in .NET 4.6.1. I expect command like Update-Database, but no. What is the right way?

**Comment by fubar-coder** [comment, marked as answer]
You should use one of the external migration runners. In your case, I'd suggest using the [`Migrate.exe` tool](https://fluentmigrator.github.io/articles/runners/runner-console.html) from the [FluentMigrator.Console nuget package](https://www.nuget.org/packages/FluentMigrator.Console/). You need to extract the package and use the `Migrate.exe` tool from the `lib\any`, `lib\x86`, or `lib\x64` directory.

**Reply by jzabroski** [reply]
To elaborate on this answer, FluentMigrator.Console contains all FluentMigrator DLLs, whereas FluentMigrator.MSBuild does not. FluentMigrator.MSBuild is just one DLL.  As a result, you need to get the DLLs you need to run the MSBuild task by either downloading them as well and putting them in the same directory as FluentMigrator.MSBuild.dll, or effectively copy FluentMigrator.MSBuild.dll over to your corresponding FluentMigrator.Console lib directory @fubar-coder  mentions above.

**Comment by jzabroski** [comment]
tl;dr: For SQL Server, I usually use sqlcmd and pass in a script file called CreateDatabase.sql - the downside being it's not portable to other SQL engines.  The upside is its easy to customize.

Long Answer:
I have a TestSetup task in MSBuild that calls DatabaseDrop and DatabaseCreate and then DatabaseMigrate targets.  My DatabaseMigrate command uses dependency injection to get a collection of ITestData instances which I seed in a maintenance migration where the MigrationStage is AfterAll and the TagName is UnitTest.  I guess it would not hurt to include a more comprehensive example of this, but I would need to think about how to make the demo work for all database dialects.

Basically, it's a good question, but it's outside the scope of this project to create your database for you.  Part of this has to do with the fact that creating database often requires administrative privileges on the database server, and some databases require permission to create a file on a device.  In effect, these tasks have nothing to do with abstracting SQL away from the C# developer.
